2019/d01p1FuelCounterUpper.d

```diff
@@ -1,7 +1,14 @@
-unittest {
-  import std.algorithm;
+import std.algorithm;
 
+unittest {
   auto input = [12, 14, 1969, 100756];
   auto fuelTotal = input.map!"a / 3 - 2".sum;
-  assert(fuelTotal == 2 + 2 + 654 + 33583);
+  assert(fuelTotal == 2 + 2 + 654 + 33583); // 34241
+} version(unittest) {} else {
+
+void main() {
+  import std.conv, std.stdio;
+  stdin.byLineCopy.map!(to!int).map!"a / 3 - 2".sum.writeln;
```

Comment:
Standard input is read by line, each line is converted to an integer, has the mass-to-fuel function applied and the sum is taken.

I don't know yet how to explain template functions except to say that the `!` operator reveals them. New here is `to` from the standard conversions library, used to convert a string to an integer. I don't know if `byLineCopy` is needed here - there exists also `byLine` which sounds faster.

The interpretation is clear though, given the preceding unit test proving that the algorithm is correct.

Reply:
And as seen in the third commit, it just worked with the puzzle input. No fumbling required 😄